gateway/src/test/java/io/zeebe/gateway/api/util/StubbedBrokerClient.java

```diff
@@ -96,15 +79,20 @@ public void close() {}
     try {
       final RequestHandler requestHandler = requestHandlers.get(request.getClass());
       final BrokerResponse<T> response = requestHandler.handle(request);
-      if (response.isResponse()) {
-        responseConsumer.accept(response.getKey(), response.getResponse());
-      } else if (response.isRejection()) {
-        throwableConsumer.accept(new BrokerRejectionException(response.getRejection()));
-      } else if (response.isError()) {
-        throwableConsumer.accept(new BrokerErrorException(response.getError()));
-      } else {
-        throwableConsumer.accept(
-            new IllegalBrokerResponseException("Unknown response received: " + response));
+      try {
+        if (response.isResponse()) {
+          responseConsumer.accept(response.getKey(), response.getResponse());
+        } else if (response.isRejection()) {
+          throwableConsumer.accept(new BrokerRejectionException(response.getRejection()));
+        } else if (response.isError()) {
+          throwableConsumer.accept(new BrokerErrorException(response.getError()));
+        } else {
+          throwableConsumer.accept(
+              new IllegalBrokerResponseException(
+                  "Expected broker response to be either response, rejection, or error, but is neither of them []"));
+        }
+      } catch (final RuntimeException e) {
+        throwableConsumer.accept(new BrokerResponseException(e));
```

Comment:
As part of resolving conflicts with 8c065fe29, I added this catch to this method, instead of to the `sendRequestWithRetry` above, as that one now reuses all of this code.